contracts/UserAuthority.sol

```diff
@@ -0,0 +1,233 @@
+pragma solidity <0.6.0;
+
+import "./interfaces/SignaturitUserInterface.sol";
+import "./libraries/Utils.sol";
+
+/*
+Gas to deploy: 2.896.394
+*/
+
+/*
+User's role:
+0 - User
+1 - Admin
+2 - Root
+
+- Everyone that create an user is also manager for that user
+- Root is manager of all the users
+
+Admins can:
+- create users
+- create admins
+- set role, reputation and validity for managed users
+
+Admins can't:
+- set role, validity and reputation of root
+- create root users
```

Comment:
The admin never can create root accounts, the root account must be unique

Reply:
indeed is a list of actions that the admin CAN'T do.